### PR TITLE
builtins: implement ST_Points

### LINKED ISSUE #49009: geo/geomfn: implement ST_Points({geometry})
Implement `ST_Points` on arguments {geometry}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_Points.html).

_Observers: Please react to this issue if you need this functionality._

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end (your exhaustive unit tests go the above mentioned packages!).
* Ensure the documentation is regenerated by calling `make buildshort`. You can also play with it by calling `./cockroach demo --empty` afterwards.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-06-30T05:29:57Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by erikgrinaker**
Release note (sql change): Implement the geometry builtin `ST_Points`.

Closes #49009.

### CODE REVIEW

**Review by otan — COMMENTED** [review]
just a heads up -- i will probably have to get you to rebase this on top of https://github.com/cockroachdb/cockroach/pull/53446.

**Comment by otan** [comment]
you'll also need to add a `Release justification` at this stage in your commit message and PR description -- builtins fall under "low risk, high benefit" changes.

We're in a period of stability right now, but the branch has not been cut yet.

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 1]
```diff
@@ -1775,6 +1775,23 @@ Flags shown square brackets after the geometry type have the following meaning:
...
+			func(ctx *tree.EvalContext, g *tree.DGeometry) (tree.Datum, error) {
+				points, err := geomfn.Points(g.Geometry)
+				if err != nil {
+					return nil, err
+				}
+				return tree.NewDGeometry(points), nil
+			},
+			types.Geometry,
+			infoBuilder{
+				info: "Returns all coordinates in the given Geometry as a multipoint, including duplicates.",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:1790]: can you name this MultiPoint?

**Comment by erikgrinaker** [comment]
Sure, I'll update this once #53446 lands.

File: pkg/geo/geomfn/unary_operators_test.go [hunk 2]
```diff
@@ -123,3 +124,59 @@ func TestDimension(t *testing.T) {
...
+		{"GEOMETRYCOLLECTION EMPTY", "MULTIPOINT EMPTY"},
+		{"GEOMETRYCOLLECTION (GEOMETRYCOLLECTION EMPTY)", "MULTIPOINT EMPTY"},
+		{
+			`GEOMETRYCOLLECTION (
+				LINESTRING(1 1, 2 2),
+				POINT(1 1),
+				GEOMETRYCOLLECTION(
+					MULTIPOINT(2 2, 3 3),
+					GEOMETRYCOLLECTION EMPTY,
+					POINT(1 1),
```
Comment (otan) [inline pkg/geo/geomfn/unary_operators_test.go:154]: can you add a test case of `MULTIPOLYGON(EMPTY, ((...)), EMPTY)`

**Comment by otan** [comment]
It has landed!

**Comment by erikgrinaker** [comment]
Test failure seems unrelated:

```
[13:22:47]Run local roachtests (13m:45s)
[13:32:50][Run local roachtests] I200827 13:32:51.212715 21802 util/binfetcher/binfetcher.go:167  [w0] downloading https://binaries.cockroachdb.com/cockroach-v20.1.4.linux-amd64.tgz?ci=true to /go/src/github.com/cockroachdb/cockroach/artifacts/cockroach-v20.1.4.linux-amd64
[13:35:49][Run local roachtests] The service message is invalid because it does not end with ] character: ##teamcity[testFailed name='cdc/bank' details='test artifacts and logs in: artifacts/cdc/bank/run_1|n	cluster.go:2183,cdc.go:636,cdc.go:224,cdc.go:607,test_runner.go:754: output in run_133418.400_n4_for_i_in: /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:4 -- for i in $(seq 1 5); do curl --retry 3 --retry-delay 1 -o /tmp/confluent.tar.gz https://storage.googleapis.com/cockroach-fixtures/tools/confluent-oss-4.0.0-2.11.tar.gz && break |||| sleep 15; done && tar xvf /tmp/confluent.tar.gz -C /tmp/confluent returned: exit status 20|n		(1) attached stack trace|n		  -- stack trace:|n		  || main.(*cluster).RunE|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2261|n		  || main.(*cluster).Run|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2181|n		  || main.kafkaManager.install|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:636|n		  || main.runCDCBank|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224|n		  || main.registerCDC.func8|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:607|n		  || main.(*testRunner).runTest.func2|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:754|n		  || runtime.goexit|n		  || 	/usr/local/go/src/runtime/asm_amd64.s:1357|n		Wraps: (2) output in run_133418.400_n4_for_i_in|n		Wraps: (3) /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:4 -- for i in $(seq 1 5); do curl --retry 3 --retry-delay 1 -o /tmp/confluent.tar.gz https://storage.googleapis.com/cockroach-fixtures/tools/confluent-oss-4.0.0-2.11.tar.gz && break |||| sleep 15; done && tar xvf /tmp/confluent.tar.gz -C /tmp/confluent returned|n		  || stderr:|n		  || Speed   Time    Time     Time  Current|n		  ||                                  Dload  Upload   Total   Spent    Left  Speed|n		  || 
[13:35:49][Run local roachtests] The service message is invalid because it does not end with ] character: ##teamcity[testFailed name='cdc/bank' details='test artifacts and logs in: artifacts/cdc/bank/run_1|n	cluster.go:2183,cdc.go:636,cdc.go:224,cdc.go:607,test_runner.go:754: output in run_133418.400_n4_for_i_in: /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:4 -- for i in $(seq 1 5); do curl --retry 3 --retry-delay 1 -o /tmp/confluent.tar.gz https://storage.googleapis.com/cockroach-fixtures/tools/confluent-oss-4.0.0-2.11.tar.gz && break |||| sleep 15; done && tar xvf /tmp/confluent.tar.gz -C /tmp/confluent returned: exit status 20|n		(1) attached stack trace|n		  -- stack trace:|n		  || main.(*cluster).RunE|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2261|n		  || main.(*cluster).Run|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2181|n		  || main.kafkaManager.install|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:636|n		  || main.runCDCBank|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:224|n		  || main.registerCDC.func8|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cdc.go:607|n		  || main.(*testRunner).runTest.func2|n		  || 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:754|n		  || runtime.goexit|n		  || 	/usr/local/go/src/runtime/asm_amd64.s:1357|n		Wraps: (2) output in run_133418.400_n4_for_i_in|n		Wraps: (3) /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:4 -- for i in $(seq 1 5); do curl --retry 3 --retry-delay 1 -o /tmp/confluent.tar.gz https://storage.googleapis.com/cockroach-fixtures/tools/confluent-oss-4.0.0-2.11.tar.gz && break |||| sleep 15; done && tar xvf /tmp/confluent.tar.gz -C /tmp/confluent returned|n		  || stderr:|n		  || Speed   Time    Time     Time  Current|n		  ||                                  Dload  Upload   Total   Spent    Left  Speed|n		  || 
[13:36:30][Run local roachtests] Error: some tests failed
[13:36:30][Run local roachtests] Process exited with code 1
```

**Review by otan — APPROVED** [review]

**Pull request merged**